src/statmech.jl

```diff
@@ -13,31 +13,25 @@ arguments:
 - `ionization_energies` is a Dict mapping elements to their first three ionization energies
 - `partition_funcs` is a Dict mapping species to their partition functions
 """
-function saha_ion_weights(T, nₑ, atom::String, ionization_energies::Dict, partition_funcs::Dict)
+function saha_ion_weights(T, nₑ, atom, ionization_energies, partition_funcs::Dict)
     χI, χII, χIII = ionization_energies[atom]
-    UI = partition_funcs[atom*"_I"](T)
-    UII = partition_funcs[atom*"_II"](T)
+    atom = Formula(atom)
+    UI = partition_funcs[Species(atom, 0)](T)
+    UII = partition_funcs[Species(atom, 1)](T)
 
     k = kboltz_eV
     transU = translational_U(electron_mass_cgs, T)
     
-    wII =  (2.0/nₑ * (UII/UI) * transU * exp(-χI/k/T))
-    wIII = if atom == "H"
+    wII =  2.0/nₑ * (UII/UI) * transU * exp(-χI/(k*T))
+    wIII = if atom == Formula(1)
```

Comment:
Feel free to ignore this if you want. But, I think this may make things more clear
```suggestion
    wIII = if atom == Formula(1) # hydrogen
```